pkg/constraints/constraint.go

```diff
@@ -29,11 +29,14 @@ type Constraint struct {
 	// https://github.com/operator-framework/enhancements/blob/master/enhancements/compound-bundle-constraints.md
 	All *CompoundConstraint `json:"all,omitempty" yaml:"all,omitempty"`
 	Any *CompoundConstraint `json:"any,omitempty" yaml:"any,omitempty"`
-	// A note on Not: this constraint isn't particularly useful by itself.
+	// A note on Not: this constraint is not particularly useful by itself.
 	// It should be used within an All constraint alongside some other constraint type
 	// since saying "do not use any of these GVKs/packages/etc." without an alternative
-	// doesn't make sense.
+	// doesn't make sense. Both Not and None can be used for this type but None is
+	// deprecated.
 	Not *CompoundConstraint `json:"not,omitempty" yaml:"not,omitempty"`
+	// Deprecated: Use Not instead. This has been deprecated as Not has a clearer UX.
```

Comment:
nit: go-doc comments on exported fields must start with the field name:

```go
// None is a...
// Deprecated: Use Not instead...
None *CompoundConstraint `json:...`
```

Reply:
Should this also be updated for the other fields?